Std/Data/AssocList.lean

```diff
@@ -3,7 +3,7 @@ Copyright (c) 2019 Microsoft Corporation. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Leonardo de Moura, Mario Carneiro
 -/
-import Std.Data.List.Basic
+import Std.Data.List.Lemmas
```

Comment:
I'm not a fan of breaking the Basic/Lemmas split this way. This suggests that it's time to split `AssocList` into Basic/Lemmas.